File: backend/models/notification.py
from models import db
from datetime import datetime

class Notification(db.Model):
    __tablename__ = 'notifications'
    
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    type = db.Column(db.String(50), nullable=False)  # task_assigned/message/comment/deadline/mention
    title = db.Column(db.String(200), nullable=False)
    content = db.Column(db.Text)
    link = db.Column(db.String(500))  # 點擊後導向的連結
    is_read = db.Column(db.Boolean, default=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # 關聯
    user = db.relationship('User', backref=db.backref('notifications', lazy=True))
    
    def __repr__(self):
        return f'<Notification {self.id} for User {self.user_id}>'

File: frontend/src/types/notification.ts
export type NotificationType =
  | 'task_assigned'
  | 'timeline_invited'
  | 'comment'
  | 'deadline'
  | 'mention';

export interface Notification {
  id: number;
  type: NotificationType;
  title: string;
  content: string | null;
  link: string | null;
  is_read: boolean;
  user_id?: number;
  created_at: string | null;
}
